sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/ValueStateImplWithTTL.scala

```diff
@@ -44,11 +44,12 @@ class ValueStateImplWithTTL[S](
     valEncoder: Encoder[S],
     ttlConfig: TTLConfig,
     batchTimestampMs: Long,
+    avroSerde: Option[AvroSerde],
     metrics: Map[String, SQLMetric] = Map.empty)
   extends SingleKeyTTLStateImpl(
     stateName, store, keyExprEnc, batchTimestampMs) with ValueState[S] {
 
-  private val stateTypesEncoder = StateTypesEncoder(keyExprEnc, valEncoder,
+  private val stateTypesEncoder = UnsafeRowTypesEncoder(keyExprEnc, valEncoder,
```

Comment:
lets just comment that avro is not supported in TTL yet

maybe better to even fail the query explicitly for now